bftengine/src/bftengine/DbCheckpointManager.cpp

```diff
@@ -199,15 +184,17 @@ void DbCheckpointManager::initializeDbCheckpointManager(std::shared_ptr<concord:
     init();
   } else {
     // db checkpoint is disabled. Cleanup metadata and checkpoints created if any
-    auto ret = std::async(std::launch::async, [this]() { cleanUp(); });
+    cleanUpFuture_ = std::async(std::launch::async, [this]() { cleanUp(); });
   }
 }
 
-void DbCheckpointManager::createDbCheckpointAsync(const SeqNum& seqNum, const std::optional<Timestamp>& timestamp) {
-  auto ret = std::async(std::launch::async, [this, seqNum, timestamp]() -> void {
-    const auto& lastBlockid = getLastBlockIdCb_();
+uint64_t DbCheckpointManager::createDbCheckpointAsync(const SeqNum& seqNum, const std::optional<Timestamp>& timestamp) {
+  const auto lastBlockid = getLastBlockIdCb_();
+  auto ret = std::async(std::launch::async, [this, seqNum, timestamp, lastBlockid]() -> void {
     createDbCheckpoint(lastBlockid, lastBlockid, seqNum, timestamp);
   });
+  dbCreateCheckPtFuture_.emplace(std::make_pair(lastBlockid, std::move(ret)));
+  return lastBlockid;
 }
```

Comment:
I think it is not directly related to this PR, but I was wondering what is going to happen if the caller calls this method twice, in succession, such that `lastBlockid` is the same in the two calls? Would that be a problem?

Reply:
May be we can skip db checkpoint creation if last created/in-prog db-checkpoint has same lastBlockId as the current one(returned from the recent lastBlockId api call). I can add this check. wdyt?

Reply:
We could do that, but I think the user needs to somehow be aware that this has happened, because, otherwise, he/she calls `create` and nothing happens. It needs to be explicit, I think. Or maybe we can add an overload that might or might not create and one that always creates.

Reply:
What I mean is that is an inherent race between the callers and the `createDbCheckpoint()` call and we might want to make the interface such that it cannot be misused.